dawn/src/dawn/Validator/IntegrityChecker.cpp

```diff
@@ -90,7 +94,21 @@ void IntegrityChecker::visit(const std::shared_ptr<iir::AssignmentExpr>& expr) {
     throw SemanticError("Attempt to assign constant expression " + value, metadata_.getFileName(),
                         expr->getSourceLocation().Line);
   }
-  ast::ASTVisitorForwarding::visit(expr);
+
+  int oldDim = curDimensions_;
+  expr->getLeft()->accept(*this);
+  int leftDim = curDimensions_;
+  expr->getRight()->accept(*this);
+  int rightDim = curDimensions_;
+
+  // we leave the unstrucutred world alone for now
+  if(instantiation_->getIIR()->getGridType() == ast::GridType::Unstructured &&
+     !dimensionsCompatible(leftDim, rightDim)) {
+    throw SemanticError("trying to assign " + std::to_string(leftDim) + "d field to " +
+                        std::to_string(rightDim) + "d field!");
```

Comment:
```suggestion
    auto dimToStr = [](int d) {
        switch(d) {
        case 1:
            return "vertical";
        case 2:
            return "horizontal";
        case 3:
            return "full";
        default:
            return "";
        }
    };
    throw SemanticError("trying to assign " + dimToStr(leftDim) + " field to " +
                        dimToStr(rightDim) + " field!");
```
Also here same ambiguity concern when reporting errors, as in the other PR.